### PR TITLE
Plane: Allow a ICE start even if disable while disarmed is set

### PR DESCRIPTION
**Pull request by WickedShell**
This allows you to bypass the disable starting while disarmed option in #25053 once. The workflow this enables is as follows:

1. Start with a disarmed/safety enabled aircraft
2. Disable the safety switch (physically or with software)
3. Preform inspections of the fixed wing flight control surfaces/motor test as desired
4. Send the MAVLink command (or jump via the mission) to allow starting while disarmed
5. Start the engine, preform engine checks
6. Arm the aircraft and start the flight.

The main target here is the minimization of risk as we move through the system. When inspecting the fixed wing surfaces we don't want to risk starting the ICE engine, but when we get to starting it we'd actually like to be disarmed, as that stops us from risking taking off by accident or flying the mission.

This had been tested before, but it has since been cherry-picked and rebased, so I need to actually retest this with hardware. This needs an accompanying MAVLink commit to define the usage of param4. (I'm planning to use it as an options bit mask). 

Accompanying MAVLink PR is https://github.com/ArduPilot/mavlink/pull/331

### CODE REVIEW

File: libraries/AP_ICEngine/AP_ICEngine.cpp [hunk 1]
```diff
@@ -468,11 +475,15 @@ bool AP_ICEngine::throttle_override(float &percentage, const float base_throttle
...
 */
-bool AP_ICEngine::engine_control(float start_control, float cold_start, float height_delay)
+bool AP_ICEngine::engine_control(float start_control, float cold_start, float height_delay, float allow_disarmed)
 {
     if (!enable) {
         return false;
     }
+
+    // always update the start while disarmed flag
+    allow_single_start_while_disarmed = is_equal(allow_disarmed, 1.0f);
```
Comment (WickedShell) [inline libraries/AP_ICEngine/AP_ICEngine.cpp:485]: This needs to become a options/flag.

File: libraries/AP_Mission/AP_Mission.h [hunk 2]
```diff
@@ -203,6 +203,7 @@ class AP_Mission
         bool start_control; // start or stop engine
         bool cold_start; // use cold start procedure
         uint16_t height_delay_cm; // height delay for start
+        bool allow_disarmed_start; // allow starting the engine while disarmed
```
Comment (WickedShell) [inline libraries/AP_Mission/AP_Mission.h:206]: Lets just consider moving this straight to a 16 bit flag, rather then keeping this minimal bool.

**Review by WickedShell — COMMENTED** [review]
Tagged some things I think need to be fixed, but wanted to get some general feedback.

**Comment by tridge** [comment]
@WickedShell doesn't build

**Comment by WickedShell** [comment]
Rebased, pulled the MAVLink change in, tested with sending the message from a GCS, as well as through a mission.

**Review by tridge — APPROVED** [review]

**Pull request merged**